lib/basedriver/image-helpers.js

```diff
@@ -1,32 +1,69 @@
-import {PNG} from 'pngjs';
+import { Buffer } from 'buffer';
+import { PNG } from 'pngjs';
+import B from 'bluebird';
 
-function cropBase64Image (base64EncodedImage, rect) {
-  let pngImage = PNG.sync.read(Buffer.from(base64EncodedImage, 'base64'));
+async function createImageFromBase64 (base64EncodedImage) {
+  const buffer = Buffer.from(base64EncodedImage, 'base64');
+  return await createImageFromBuffer(buffer);
+}
+
+async function createImageFromBuffer (imageBuffer) {
+  return new B((resolve, reject) => {
+    const image = new PNG({filterType: 4});
+    image.parse(imageBuffer, (err, image) => { // eslint-disable-line promise/prefer-await-to-callbacks
+      if (err) { return reject(err); }
+      resolve(image);
+    });
+  });
+}
+
+async function createBase64FromImage (image) {
+  return new B((resolve, reject) => {
+    const chunks = [];
+    image.pack().on('data', (chunk) => chunks.push(chunk)).on('end', () => {
+      resolve(Buffer.concat(chunks).toString('base64'));
+    }).on('error', (err) => { // eslint-disable-line promise/prefer-await-to-callbacks
+      reject(err);
+    });
+  });
+}
 
-  if (rect.top < 0 || rect.top >= pngImage.height || rect.left < 0 || rect.left >= pngImage.width) {
+function cropImage (image, rect) {
+  const imageRect = {width: image.width, height: image.height};
+  const intersectionRect = getRectIntersection(rect, imageRect);
+  if (!isValidRect(intersectionRect)) {
```

Comment:
is it expected that intersectionRect variable is only used to check if the intersection is valid and is not used for further calculations at all?

Reply:
No, it is mistake